tools/triage/filters.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import json
+
+from .add_triage_signature import ALL_SIGNATURES
+from tools.jira_client import JiraAPI
+
+SIGNATURE_BY_TYPE = {signature_class.__name__.lower(): signature_class for signature_class in ALL_SIGNATURES}
+
+logger = logging.getLogger(__name__)
+
+
+class TriageIssue:
+    def __init__(self, issue, root_issue, message):
+        self.issue = issue
+        self.root_issue = root_issue
+        self.message = message
+
+
+class Filter:
+    def __init__(self, identifier, root_issue, message):
+        self.identifier = identifier
+        self.message = message
+        self.root_issue = root_issue
+
+    def matches(self, issue):
+        return issue.fields.status.name not in ("Closed", "Done", "Obsolete")
+
+    def __str__(self):
+        root_issue_str = self.root_issue.key or None
+        return f"identifier={self.identifier} root_issue={root_issue_str} message={self.message}"
+
+    def __eq__(self, other):
+        return (
+            self.identifier == other.identifier
+            and self.message == other.message
+            and (self.root_issue.key or None) == (other.root_issue.key or None)
+        )
+
+
+class LabelFilter(Filter):
+    def matches(self, issue):
+        if not super().matches(issue):
+            return False
+        try:
+            for label in issue.fields.labels:
+                if label == self.identifier:
+                    return True
+        except AttributeError:
+            return False
+        return False
+
+
+class SignatureFilter(Filter):
+    def __init__(self, jira_client, identifier, root_issue, message):
+        super().__init__(identifier, root_issue, message)
+        self._signature_class_name = self._get_signature_class_name(identifier)
+        root_issue_key = None
+        if root_issue:
+            root_issue_key = root_issue.key
+        self.signature_instance = self._signature_class_name(jira_client, root_issue_key)
+        self.jira_api = JiraAPI(jira_client, logger)
+
+    def _get_signature_class_name(self, signature_class_alias_name):
+        signature_class_alias_name = signature_class_alias_name.lower().replace("-", "").replace("_", "")
+        signature_class_name = SIGNATURE_BY_TYPE.get(signature_class_alias_name)
+        assert signature_class_name is not None
+        return signature_class_name
```

Comment:
pretty much does the same
```suggestion
        return SIGNATURE_BY_TYPE[signature_class_alias_name]
```